packages/python/javdb_platform/db.py

```
@@ -1818,21 +2552,441 @@ def db_create_report_session(
+    if current_status == 'committed' and not force:
+        raise ValueError(
+            f"Refusing to roll back ReportSessions.Id={session_id} because "
+            f"Status='committed'. Pass force=True if you really intend to "
+            f"undo a successful run's writes."
+        )
+
+    if not dry_run:
+        # Best-effort flag — failure here shouldn't block the rollback.
+        try:
+            db_mark_session_failed(session_id, db_path=reports_db_path)
+        except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)